src/wtc-gallery-component.js

```diff
@@ -321,6 +321,38 @@ class Gallery extends ElementController {
     return this;
   }
 
+  /**
+   * Handles the setting of the timer for autoplay galleries,
+   * taking into consideration the document's visibility state.
+   */
+  handleAutoplay() {
+    if (!this.options.autoplay) return;
+
+    if (document.visibilityState === "hidden") {
+      this.pause();
+      document.addEventListener(
+        "visibilitychange",
+        this.handleVisibilityChange.bind(this)
+      );
+    } else {
+      this.player = setTimeout(this.next.bind(this), this.options.delay);
+    }
+  }
+
+  /**
+   * Fires when the document's visbility chnages from the "hidden" state;
+   * Resumes autoplay functionality if applicable.
+   */
+  handleVisibilityChange() {
+    if (!this.options.autoplay || document.visibilityState === "hidden") return;
+
+    document.removeEventListener(
+      "visibilitychange",
+      this.handleVisibilityChange.bind(this)
```

Comment:
Are you sure this works? I had thought that removeEventListener only worked if you supplied it with the exact function that you supplied to the addEventListener. Technically this is a different function as bing acts as a wrapper around the original function.

Reply:
great shout. We should bind it in the constructor

Reply:
Fixed! @liamegan